src/index.js

```diff
@@ -13,9 +13,9 @@ const useEventListener = (eventName, handler, element = global) => {
       if (!isSupported) return;
 
       const eventListener = event => savedHandler.current(event);
-      element.addEventListener(eventName, eventListener);
+      element.addEventListener(eventName, eventListener, useCapture);
       return () => {
-        element.removeEventListener(eventName, eventListener);
+        element.removeEventListener(eventName, eventListener, useCapture);
       };
     },
     [eventName, element]
```

Comment:
we should check whether options is an object. If it is, destructure it to `capture, once, passive`.
then `const optionsOrUseCaptureMemo = useMemo(() => optionsOrUseCapture, [capture, once, passive])` to guard against creating new objects
and pass the optionsOrUseCaptureMemo to this dependency array